.github/workflows/proto-docker.yml

```diff
@@ -31,16 +31,28 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
 
-      - name: Login to DockerHub
-        uses: docker/login-action@v2
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2 
+        if: ${{ github.event_name != 'pull_request' }}
         with:
-          username: ${{ secrets.DOCKERHUBTM_USERNAME }}
-          password: ${{ secrets.DOCKERHUBTM_TOKEN }}
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          
+      - name: Build
+        uses: docker/build-push-action@v2
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          file: contrib/devtools/Dockerfile
+          context: ./contrib/devtools
+          platforms: linux/amd64,linux/arm64
+          push: false
```

Comment:
why this?

Reply:
If someone makes changes to the Dockerfile that aren't in the main branch, it will build, but not push.

Reply:
removed